compiler/rustc_macros/src/query.rs

```diff
@@ -272,6 +272,34 @@ fn process_modifiers(query: &mut Query) -> QueryModifiers {
                 if desc.is_some() {
                     panic!("duplicate modifier `desc` for query `{}`", query.name);
                 }
+                if query.doc_comments.is_empty() {
+                    use ::syn::*;
+                    let mut list = list.iter();
+                    let format_str: String = match list.next() {
+                        Some(&Expr::Lit(ExprLit { lit: Lit::Str(ref lit_str), .. })) => {
+                            lit_str.value().replace("`{}`", "{}") // We add them later anyways for consistency
+                        }
+                        _ => panic!("Expected a string literal"),
```

Comment:
`bug!` is usually used instead of `panic!` where possible (maybe you could change the `panic` on line 273 too).

Reply:
Note that this is proc-macro code though; isn't `bug!` intended for ICEs? I don't think this code can ICE because it's run while compiling *rustc*.

Reply:
bug is defined in rustc_middle, which itself depends on this crate. So I don't think there's a nice way to do this; FWIW other errors also use `panic!`: https://github.com/rust-lang/rust/blob/8a5022438fa8a882f301c43e18cb5ad8ae640ec3/compiler/rustc_macros/src/query.rs#L255